components/script/dom/htmltablecellelement.rs

```diff
@@ -174,23 +178,26 @@ impl VirtualMethods for HTMLTableCellElement {
         match *local_name {
             local_name!("colspan") => {
                 let mut attr = AttrValue::from_u32(value.into(), DEFAULT_COLSPAN);
-                if let AttrValue::UInt(_, ref mut val) = attr {
-                    if *val == 0 {
-                        *val = 1;
-                    }
+                if let AttrValue::UInt(_, ref mut value) = attr {
+                    // From <https://html.spec.whatwg.org/multipage/#dom-tdth-colspan>:
+                    // > The colSpan IDL attribute must reflect the colspan content attribute. It is clamped to
+                    // > the range [1, 1000], and its default value is 1.
+                    *value = (*value).clamp(1, 1000);
                 }
                 attr
             },
             local_name!("rowspan") => {
                 let mut attr = AttrValue::from_u32(value.into(), DEFAULT_ROWSPAN);
-                if let AttrValue::UInt(_, ref mut val) = attr {
-                    if *val == 0 {
-                        let node = self.upcast::<Node>();
-                        let doc = node.owner_doc();
-                        // rowspan = 0 is not supported in quirks mode
-                        if doc.quirks_mode() != QuirksMode::NoQuirks {
-                            *val = 1;
-                        }
+                if let AttrValue::UInt(_, ref mut value) = attr {
+                    // From <https://html.spec.whatwg.org/multipage/#dom-tdth-rowspan>:
+                    // > The rowSpan IDL attribute must reflect the rowspan content attribute. It is clamped to
+                    // > the range [0, 65534], and its default value is 1.
+                    // Note that rowspan = 0 is not supported in quirks mode.
+                    let document = self.upcast::<Node>().owner_doc();
+                    if document.quirks_mode() != QuirksMode::NoQuirks {
```

Comment:
Existing problem, but this should be `quirks_mode() == QuirksMode::Quirks`.

Firefox says 0 in limited-quirks mode:

```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Frameset//EN"
   "http://www.w3.org/TR/html4/frameset.dtd">
<table><td id="foo" rowspan="0"></td></table>
<script>document.write(document.getElementById('foo').rowSpan);</script>
```

I see a bunch of `QuirksMode::NoQuirks` in the code that may need to be double-checked, so can be left for a follow-up.